app/models/events/user_mentioned.rb

```diff
@@ -1,18 +1,14 @@
-class Events::UserMentioned < Event
-  include Events::Notify::InApp
-  include Events::Notify::ByEmail
-
-  def self.publish!(model, actor, mentioned_user)
-    super model, user: actor, custom_fields: { mentioned_user_id: mentioned_user.id }
+class Events::UserMentioned < Events::AnnouncementCreated
```

Comment:
Cool, Nice idea.

mentioned_in looks for notifications hanging off previously created user_mentioned events, so we have a few problems with them being kind: announcement_created, which can be solved by them being user_mentioned.

- switching tracks half way, editing previous comments will re-mention because of the change to looking for announcement_created from user_mentioned in mentioned_in
- we have a database that at a certain point in history, has a completely different method of mentioning, which may haunt us.
- desired behaviour for mentioning vs announcing is different. if you've already announced a poll, then you mention someone in it later, unless they have different kinds, the announcement will look like a mention to mentioned_in, so they wont receive the mention, even though you could have re announced it to them.

Plus I think it is nice to prove that this code is really decoupled and we can use it for more than one event kind.